control-plane/pkg/reconciler/source/v2/sourcev2.go

```diff
@@ -0,0 +1,136 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	sources "knative.dev/eventing-kafka/pkg/apis/sources/v1beta1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/reconciler"
+
+	internals "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/internals/kafka/eventing"
+	internalscg "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1"
+	internalsclient "knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/clientset/versioned"
+	internalslst "knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/listers/eventing/v1alpha1"
+)
+
+const (
+	DefaultDeliveryOrder = internals.Ordered
+
+	KafkaConditionConsumerGroup apis.ConditionType = "ConsumerGroup" //condition is registered by controller
+)
+
+var (
+	conditionSet = apis.NewLivingConditionSet(
+		KafkaConditionConsumerGroup,
+	)
+)
+
+type ReconcilerV2 struct {
+	ConsumerGroupLister internalslst.ConsumerGroupLister
+	InternalsClient     internalsclient.Interface
+}
+
+func (r *ReconcilerV2) ReconcileKind(ctx context.Context, ks *sources.KafkaSource) reconciler.Event {
+
+	consgroup, err := r.reconcileConsumerGroup(ctx, ks)
+	if err != nil {
+		ks.GetConditionSet().Manage(&ks.Status).MarkFalse(KafkaConditionConsumerGroup, "failed to reconcile consumer group", err.Error())
+		return err
+	}
+	ks.GetConditionSet().Manage(&ks.Status).MarkTrue(KafkaConditionConsumerGroup)
+
+	ks.Status.MarkSink(consgroup.Status.SubscriberURI)
+	ks.Status.Placeable = consgroup.Status.Placeable
+	ks.Status.Consumers = *consgroup.Status.Replicas
+
+	return nil
+}
+
+func (r ReconcilerV2) reconcileConsumerGroup(ctx context.Context, ks *sources.KafkaSource) (*internalscg.ConsumerGroup, error) {
+
+	newcg := &internalscg.ConsumerGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      string(ks.UID),
+			Namespace: ks.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*kmeta.NewControllerRef(ks),
+			},
+		},
+		Spec: internalscg.ConsumerGroupSpec{
+			Template: internalscg.ConsumerTemplateSpec{
+				Spec: internalscg.ConsumerSpec{
+					Topics: ks.Spec.Topics,
+					Configs: internalscg.ConsumerConfigs{Configs: map[string]string{
+						"group.id":          ks.Spec.ConsumerGroup,
+						"bootstrap.servers": strings.Join(ks.Spec.BootstrapServers, ","),
+					}},
+					Auth: &internalscg.Auth{
+						NetSpec: &ks.Spec.KafkaAuthSpec.Net,
+					},
+					Delivery: &internalscg.DeliverySpec{
+						Ordering: DefaultDeliveryOrder,
+					},
+					Subscriber: ks.Spec.Sink,
+					CloudEventOverrides: &duckv1.CloudEventOverrides{
+						Extensions: ks.Spec.CloudEventOverrides.Extensions,
+					},
+				},
+			},
+		},
```

Comment:
Shouldn't we set `spec.replicas = ks.spec.replicas` too?

Reply:
Yes! That was missed - added now.